### DISCUSSION TITLE
S3 Presigned URL getting expired within 1 hour

### DISCUSSION THREAD

**Discussion by DeweshTiwari** [aws/aws-sdk-java#3171] [Q&A]
I have certain files at S3 bucket, and generating presigned url programmatically to access those files , Even though i am setting expire time as 7 hours (setting correctly as X-Amz-Expires = 25200 ) it is getting expired after an hour .

**Comment by debora-ito** [comment]
If you are using temporary credentials to generate the presigned url, check if they expire in one hour. 

The actual expiration of a presigned url is the minimum between the credentials and the presigned url expiration time.
